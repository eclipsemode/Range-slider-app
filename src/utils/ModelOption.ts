declare type ModelOption = {
    from: number,
    to: number,
    gap: number,
    step: number,
    min: number,
    max: number,
    vertical: boolean;
    rulers: boolean,
    range: boolean,
    tooltip: boolean,
    percent: boolean,
    color: {
        firstColor: string,
        secondColor: string,
        textColor: string,
        thumbColor: string
    },
    configPanel: boolean,
    controlConfig: string[],
    toggleConfig: string[],
    progress: boolean
}

export default ModelOption;