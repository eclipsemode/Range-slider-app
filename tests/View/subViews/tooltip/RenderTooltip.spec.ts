import $ from "jquery";
import {
  CreateBar,
  CreateTooltip,
  RenderTooltip,
} from "../../../../src/View/subViews";
import ModelOption from "../../../../src/utils/ModelOption";

describe("Render Tooltip tests", () => {
  const body: JQuery = $("body");
  const bar: CreateBar = new CreateBar(body);
  const tooltip: CreateTooltip = new CreateTooltip(body);
  const options: ModelOption = {
    max: 104,
    tooltip: true,
    range: false,
    configPanel: true,
    step: 33,
    gap: 1,
    min: 0,
    from: 0,
    to: 99,
    vertical: false,
    rulers: true,
    percent: false,
    color: {
      firstColor: "#ffe53b",
      secondColor: "#ff2525",
      textColor: "#ffe53b",
      thumbColor: "#ffe53b",
    },
    progress: true,
    controlConfig: ["min", "max", "step", "from", "to"],
    toggleConfig: ["vertical", "range", "rulers", "progress", "tooltip"],
  };
  test("Should be defined", () => {
    const renderTooltip: RenderTooltip = new RenderTooltip(
      options,
      tooltip,
      tooltip,
      bar
    );
    expect(renderTooltip.tooltipFrom).toBeDefined();
    expect(renderTooltip.tooltipTo).toBeNull();
  });
  test("Should be defined tooltip TO", () => {
    const renderTooltip: RenderTooltip = new RenderTooltip(
      { ...options, range: true },
      tooltip,
      tooltip,
      bar
    );
    expect(renderTooltip.tooltipTo).toBeDefined();
  });
  test("Should match snapshot", () => {
    const renderTooltip: RenderTooltip = new RenderTooltip(
      { ...options, range: true },
      tooltip,
      tooltip,
      bar
    );
    expect(
      renderTooltip.tooltipFrom.tooltipElement[0].innerHTML
    ).toMatchSnapshot('class="slider-app__bar"');
  });
  test("Should match className", () => {
    const renderTooltip: RenderTooltip = new RenderTooltip(
      { ...options, range: true, vertical: true },
      tooltip,
      tooltip,
      bar
    );
    expect(renderTooltip.tooltipFrom.tooltipElement[0].className).toMatch(
      "slider-app__tooltip--vertical"
    );
  });
});
