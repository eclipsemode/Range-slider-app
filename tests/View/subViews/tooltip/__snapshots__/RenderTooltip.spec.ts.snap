// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Tooltip tests Should match snapshot: cascasxca 1`] = `
jQuery {
  "0": <body>
    <div
      class="slider-app__bar"
    />
  </body>,
  "length": 1,
  "prevObject": jQuery {
    "0": Document {
      "location": Location {
        "assign": [Function],
        "hash": "",
        "host": "localhost",
        "hostname": "localhost",
        "href": "http://localhost/",
        "origin": "http://localhost",
        "pathname": "/",
        "port": "",
        "protocol": "http:",
        "reload": [Function],
        "replace": [Function],
        "search": "",
        "toString": [Function],
      },
      Symbol(SameObject caches): Object {
        "childNodes": NodeList [
          DocumentType {},
          <html>
            <head />
            <body>
              <div
                class="slider-app__bar"
              />
            </body>
          </html>,
        ],
        "implementation": DOMImplementation {},
      },
    },
    "length": 1,
  },
}
`;

exports[`Render Tooltip tests Should match snapshot: class="slider-app__bar" 1`] = `"0 - 99"`;

exports[`Render Tooltip tests Should match snapshot: dasfasfas 1`] = `
RenderTooltip {
  "bar": CreateBar {
    "barElement": jQuery {
      "0": <div
        class="slider-app__bar"
      />,
      "length": 1,
    },
    "parent": jQuery {
      "0": <body>
        <div
          class="slider-app__bar"
        />
      </body>,
      "length": 1,
      "prevObject": jQuery {
        "0": Document {
          "location": Location {
            "assign": [Function],
            "hash": "",
            "host": "localhost",
            "hostname": "localhost",
            "href": "http://localhost/",
            "origin": "http://localhost",
            "pathname": "/",
            "port": "",
            "protocol": "http:",
            "reload": [Function],
            "replace": [Function],
            "search": "",
            "toString": [Function],
          },
          Symbol(SameObject caches): Object {
            "childNodes": NodeList [
              DocumentType {},
              <html>
                <head />
                <body>
                  <div
                    class="slider-app__bar"
                  />
                </body>
              </html>,
            ],
            "implementation": DOMImplementation {},
          },
        },
        "length": 1,
      },
    },
  },
  "options": Object {
    "color": Object {
      "firstColor": "#ffe53b",
      "secondColor": "#ff2525",
      "textColor": "#ffe53b",
      "thumbColor": "#ffe53b",
    },
    "configPanel": true,
    "controlConfig": Array [
      "min",
      "max",
      "step",
      "from",
      "to",
    ],
    "from": 0,
    "gap": 1,
    "max": 104,
    "min": 0,
    "percent": false,
    "progress": true,
    "range": true,
    "rulers": true,
    "step": 33,
    "to": 99,
    "toggleConfig": Array [
      "vertical",
      "range",
      "rulers",
      "progress",
      "tooltip",
    ],
    "tooltip": true,
    "vertical": false,
  },
  "tooltipFrom": CreateTooltip {
    "parent": jQuery {
      "0": <body>
        <div
          class="slider-app__bar"
        />
      </body>,
      "length": 1,
      "prevObject": jQuery {
        "0": Document {
          "location": Location {
            "assign": [Function],
            "hash": "",
            "host": "localhost",
            "hostname": "localhost",
            "href": "http://localhost/",
            "origin": "http://localhost",
            "pathname": "/",
            "port": "",
            "protocol": "http:",
            "reload": [Function],
            "replace": [Function],
            "search": "",
            "toString": [Function],
          },
          Symbol(SameObject caches): Object {
            "childNodes": NodeList [
              DocumentType {},
              <html>
                <head />
                <body>
                  <div
                    class="slider-app__bar"
                  />
                </body>
              </html>,
            ],
            "implementation": DOMImplementation {},
          },
        },
        "length": 1,
      },
    },
    "tooltipElement": jQuery {
      "0": <div
        class="slider-app__tooltip slider-app__tooltip--merged"
        style="left: 0px; bottom: 2.2em;"
      >
        0 - 99
      </div>,
      "length": 1,
    },
  },
  "tooltipTo": null,
}
`;

exports[`Render Tooltip tests Should match snapshot: slider-app__tooltip 1`] = `
jQuery {
  "0": <body>
    <div
      class="slider-app__bar"
    />
  </body>,
  "length": 1,
  "prevObject": jQuery {
    "0": Document {
      "location": Location {
        "assign": [Function],
        "hash": "",
        "host": "localhost",
        "hostname": "localhost",
        "href": "http://localhost/",
        "origin": "http://localhost",
        "pathname": "/",
        "port": "",
        "protocol": "http:",
        "reload": [Function],
        "replace": [Function],
        "search": "",
        "toString": [Function],
      },
      Symbol(SameObject caches): Object {
        "childNodes": NodeList [
          DocumentType {},
          <html>
            <head />
            <body>
              <div
                class="slider-app__bar"
              />
            </body>
          </html>,
        ],
        "implementation": DOMImplementation {},
      },
    },
    "length": 1,
  },
}
`;
